### PR TITLE
Implement automated lintr checks

### LINKED ISSUE #63: [FEATURE] Implement automated lintr and styler checks
# Feature Request Description
We should have automated checked to ensure the package is lint free and follows appropriate style

# Proposed Solution
For `lintr` we can add an `expect_lint_free()` test to our testing suite. This will give us a full lint check without changing GHA workflows.

For `styler` the best solution is probably [`precommit`](https://lorenzwalthert.github.io/precommit/articles/available-hooks.html#style-files). The hook can be configured to style files automatically or just alert for failures without modifying files.

### Checklist

- [x] The issue is atomic
- [x] The issue description is documented
- [x] The issue title describes the problem succinctly
- [x] Developers are assigned to the issue
- [x] Labels are assigned to the issue

### PR DESCRIPTION
**Pull request by skadauke**
# Description

This PR implements linting using lintr. It introduces a reasonable `.lintr` file and fixes lints found in the package. It also adds a `lint` GitHub Action that fails when any lints are found. Styler was also run against the files, and all style issues were fixed.

Note: [styler supports nicely aligned tribbles](https://styler.r-lib.org/articles/detect-alignment.html) but it doesn't go out of its way to align it. You have to make sure that they are correctly left aligned:

```r
# styler will leave this one alone because it's already correctly left-aligned
tibble::tribble(
  ~this, ~works,
  "a",   "b"
)

# styler will smush this to remove extra whitespaces because it's not correctly aligned
tibble::tribble(
  ~this, ~doesnt,
  "a",    "b"
)
```

# Proposed Changes 

- Add `.lintr` file that allows allows 120 characters per line and names up to 45 characters in length
- Fix various lints and add `# nolint` comments in places where linters give false positives. Note: `object_usage_linter` sometimes throws false positives within the context of `map()` alerts (https://github.com/r-lib/lintr/issues/1527). Also, even though it understands `glue` string interpolation, it doesn't appear to realize that `cli` alerts use `glue` type interpolation as well and will throw false positives there as well. I think it's fine to deal with this by using `# nolint: object_use_linter`.
- Add a the lint GHA workflow as described [here](https://lintr.r-lib.org/articles/continuous-integration.html). ~Use `LINTR_ERROR_ON_LINT = false` so that lints are reported but no CI failure is thrown.~ Apparently `lintr-bot` no longer comments on PRs so for now I have `lint` fail when any lints are found. As an alternative to the GHA workflow, we could consider adding a `expect_lint_free()` to our `testthat` suite. The downside is that it would slow down the test suite. The upside would be that lints would fail loudly and rapidly so exert more pressure on the package to stay lint free.
- ~Add the style GHA workflow described [here](https://github.com/r-lib/actions/blob/v2/examples/README.md#style-package). This workflow should automatically style all code and if there are any changes push those changes back onto the current branch automatically.~ This doesn't work. I've added `styler::style_pkg()` to `refresh.R`. That way we remember to at least periodically style the package such as e.g. before submitting to CRAN.

### Issue Addressed
**Closes #63**

### PR Checklist
Before submitting this PR, please check and verify below that the submission meets the below criteria:

- NA New/revised functions have associated tests
- NA New/revised functions that update downstream outputs have associated static testing files (`.RDS`) updated under `inst/testdata/create_test_data.R`
- NA New tests that make API calls use `httptest::with_mock_api` and any new mocks were added to `tests/testthat/fixtures/create_httptest_mocks.R`
- NA New/revised functions use appropriate naming conventions
- NA New/revised functions don't repeat code
- [x] Code changes are less than **250** lines total
- [x] Issues linked to the PR using [GitHub's list of keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] The appropriate reviewer is assigned to the PR
- [x] The appropriate developers are assigned to the PR

# Code Review
This section to be used by the reviewer and developers during Code Review after PR submission

### Code Review Checklist

- [ ] I checked that new files follow naming conventions and are in the right place
- [ ] I checked that documentation is complete, clear, and without typos
- [ ] I added/edited comments to explain "why" not "how"
- [ ] I checked that all new variable and function names follow naming conventions
- [ ] I checked that new tests have been written for key business logic and/or bugs that this PR fixes
- [ ] I checked that new tests address important edge cases

### CODE REVIEW

File: tests/testthat/test-labelled.R [hunk 1]
```diff
@@ -1,7 +1,7 @@
 
-test_that("make_labelled applies labels to all elements of supertibble",{
+test_that("make_labelled applies labels to all elements of supertibble", {
   supertbl <- tibble::tribble(
-    ~ redcap_data, ~ redcap_metadata, ~ redcap_events,
+    ~redcap_data, ~redcap_metadata, ~redcap_events,
```
Comment (rsh52) [inline tests/testthat/test-labelled.R:4]: How come here it gets rid of the space between variable and `~` but below it adds it? I guess because the application is different (variable vs function)?

File: vignettes/REDCapTidieR.Rmd [hunk 2]
```diff
@@ -215,7 +215,7 @@ The label for `name` doesn't look quite right. Let's remove that trailing `:`. T
 
 ```{r}
 superheroes |>
-  make_labelled(format_labels = ~gsub(":", "", .)) |>
+  make_labelled(format_labels = ~ gsub(":", "", .)) |>
```
Comment (rsh52) [inline vignettes/REDCapTidieR.Rmd:218]: See above `~` comment.

File: .lintr [hunk 3]
```diff
@@ -0,0 +1,4 @@
+linters: linters_with_defaults(
+    line_length_linter(120),
+    object_length_linter(45)
```
Comment (rsh52) [inline .lintr:3]: Looks good, I can't think of any others we'd want to add at the moment but I like how easy it is to add more.

**Review by rsh52 — APPROVED** [review]
LGTM, just a couple of remarks about lintr behavior. Thanks for working on this!

File: tests/testthat/test-labelled.R (hunk 1 above)
Comment (skadauke) [reply tests/testthat/test-labelled.R:4]: In the context of tribbles, there's no space after the tilde, in the context of anonymous functions there's a space.

File: R/clean_redcap_long.R [hunk 4]
```diff
@@ -26,12 +26,9 @@
...
 
-clean_redcap_long <- function(
-    db_data_long,
-    db_metadata_long,
-    linked_arms
-) {
-
+clean_redcap_long <- function(db_data_long,
+                              db_metadata_long,
+                              linked_arms) {
```
[the remark this answers is not in the document]
Comment (skadauke) [reply R/clean_redcap_long.R:31]: Yes, apparently tidyverse style is to make an exception to standard indentation for function definitions, and not for function calls. IMO, I think that makes complicated function calls much more readable.
